### PR TITLE
Add nightly test scripts

### PR DESCRIPTION
**Pull request by jordanlewis**
Add some test scripts for TeamCity. These are hooked up to the [Roachperf Nightly](https://teamcity.cockroachdb.com/viewType.html?buildTypeId=Cockroach_Nightlies_RoachperfNightly) TeamCity test and will get run every night.

I still need to have it upload the results to S3. For now it stores them in the TeamCity artifacts, which is definitely not sufficient for the long term.

### CODE REVIEW

File: teamcity-nightly.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+echo "${TEST_CLUSTER_SSH_KEY}" > id_test_cluster.rsa
```
Comment (bdarnell) [inline teamcity-nightly.sh]: Is teamcity set up to only set this env var when building from master (and not when building PRs)?

Comment (jordanlewis) [reply teamcity-nightly.sh]: Yep, agreed we should have this spin up and down a cluster in the future.

This env var is only set when running this nightly, which doesn't get triggered by PRs.

**Comment by petermattis** [comment]
> Taking over a permanent cluster for this is either error-prone (if people use it for manual testing during the daytime) or expensive (if we leave it idle during the day).

Agreed. This is a stop-gap solution.

File: teamcity-nightly.sh (hunk 1 above)
Comment (petermattis) [reply teamcity-nightly.sh]: @benesch Has suggested that spinning up and down a cluster should be integrated with `roachperf` (or perhaps a better named tool). I'm not at all familiar with terraform, but this doesn't seem like too much work to adapt from our existing terraform tools.

**Pull request merged**